test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Batch/Tests/DataServicesClient/DefaultODataBatchHandlerTests.cs

```diff
@@ -241,6 +241,150 @@ public async Task CanHandleAbsoluteAndRelativeUrls()
             Assert.Equal(3, subResponseCount);
         }
 
+
+        [Fact]
+        public async Task CanHandleAbsoluteAndRelativeUrlsJSON()
+        {
+            // Arrange
+            var requestUri = string.Format("{0}/DefaultBatch/$batch", this.BaseAddress);
+            Uri address = new Uri(this.BaseAddress, UriKind.Absolute);
+
+            string relativeToServiceRootUri = "DefaultBatchCustomer";
+            string relativeToHostUri = address.LocalPath.TrimEnd(new char[] { '/' }) + "/DefaultBatch/DefaultBatchCustomer";
+            string absoluteUri = this.BaseAddress + "/DefaultBatch/DefaultBatchCustomer";
+
+            // Act
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, requestUri);
+            request.Headers.Accept.Add(MediaTypeWithQualityHeaderValue.Parse("application/json"));
```

Comment:
>request.Headers.Accept.Add(MediaTypeWithQualityHeaderValue.Parse("application/json")); [](start = 12, length = 86)

this line is mandatory?

Reply:
In WebAPI, if you don't specify an accept header, the response is multipart/mixed.  We should maybe fix this...

---
In reply to: [377800365](https://github.com/OData/WebApi/pull/2048#discussion_r377800365) [](ancestors = 377800365)

Reply:
Updated WebAPI to set default to JSON if request is JSON and  no accept is specified.

---
In reply to: [377925870](https://github.com/OData/WebApi/pull/2048#discussion_r377925870) [](ancestors = 377925870,377800365)